### PR TITLE
 Ability to compare to null in where expressions

### PR DESCRIPTION
**Pull request by markledwich2**
Added ability to compare to null in where expressions
e.g. `t.TableQuery<Customer>().Where(c => c.Nickname == null)`

### CODE REVIEW

File: source/Nevermore.IntegrationTests/IntegrationTestDatabase.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ namespace Nevermore.IntegrationTests
     public class IntegrationTestDatabase
     {
         readonly TextWriter output = Console.Out;
-        readonly string SqlInstance = "(local)\\SQLEXPRESS,1433";
+        readonly string SqlInstance = Environment.GetEnvironmentVariable("NevermoreTestServer") ?? "(local)\\SQLEXPRESS,1433";
```
Comment (droyad) [inline source/Nevermore.IntegrationTests/IntegrationTestDatabase.cs]: I'm curious about this addition. Was it you have a default instance and it didn't get picked up by .NET Core? If so we have the `SqlAlias` library for that.

File: source/Nevermore/QueryBuilder.cs [hunk 2]
```diff
@@ -44,6 +44,12 @@ public IQueryBuilder<TRecord> Where(string whereClause)
             return this;
         }
 
+        public IQueryBuilder<TRecord> WhereNull(string fieldName, bool not)
```
Comment (droyad) [inline source/Nevermore/QueryBuilder.cs]: I would add `WhereNotNull` instead of passing a `not` flag

Comment (markledwich2) [reply source/Nevermore/QueryBuilder.cs]: Yes, that's better

File: source/Nevermore.IntegrationTests/IntegrationTestDatabase.cs (hunk 1 above)
Comment (markledwich2) [reply source/Nevermore.IntegrationTests/IntegrationTestDatabase.cs]: The /SQLEXPRESS alias redirection is esoteric. I have been ensuring that database connection dependencies are configurable through normal app settings or environment variables everywhere.

File: source/Nevermore/QueryBuilderWhereExtensions.cs [hunk 3]
```diff
@@ -146,7 +146,7 @@ static IQueryBuilder<TRecord> AddUnaryWhereClauseFromExpression<TRecord>(IQueryB
             var fieldName = property.Name;
             
             if (value == null && new[] {UnarySqlOperand.Equal, UnarySqlOperand.NotEqual}.Contains(operand))
-                return queryBuilder.WhereNull(fieldName, operand == UnarySqlOperand.NotEqual);
+                return operand == UnarySqlOperand.NotEqual ? queryBuilder.WhereNotNull(fieldName) : queryBuilder.WhereNull(fieldName);
```
Comment (droyad) [inline source/Nevermore/QueryBuilderWhereExtensions.cs]: Can I be pedantic and say inverting this to the positive would read better?

"If equal operator then call where null else call where not null" vs "If not equal operator then call where not null else call where null"

**Pull request merged**